packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -27,46 +66,87 @@ const BarrierSelector = ({ barrier_1, onChange, setHoveredBarrier, turbos_barrie
         </React.Fragment>
     );
 
-    const toggleBarriersTable = () => setIsBarriersTableExpanded(!is_barriers_table_expanded);
+    const header_mobile = (
+        <React.Fragment>
+            {localize('Barriers')}
+            <Popover
+                alignment='bottom'
+                icon='info'
+                is_bubble_hover_enabled
+                zIndex={9999}
+                message={header_tooltip_text}
+                is_open={is_mobile_tooltip_visible}
+                onClick={toggleMobileTooltip}
+            />
+        </React.Fragment>
+    );
 
-    const onBarrierClick = barrier => {
-        setHoveredBarrier(null);
-        setSelectedBarrier(barrier);
-        onChange({
-            target: {
-                name: 'barrier_1',
-                value: barrier,
-            },
-        });
-    };
+    const barriers_list = (
+        <React.Fragment>
+            <div className='trade-container__barriers-table__text'>Distance to spot</div>
+            <ThemedScrollbars height={isMobile() ? 'calc(100% - 5.5rem)' : null} autohide={false}>
+                <BarriersList
+                    base_classname='trade-container__barriers-table__item'
+                    active_item_classname='trade-container__barriers-table__item--selected'
+                    className='trade-container__barriers-table__list'
+                    list={turbos_barrier_choices}
+                    selected_item={selected_barrier}
+                    onClick={onBarrierClick}
+                    onHover={barrier => setHoveredBarrier(barrier)}
+                />
+            </ThemedScrollbars>
+        </React.Fragment>
+    );
 
-    React.useEffect(() => {
```

Comment:
I replaced it with the current code:
```
React.useEffect(() => {
        setSelectedBarrier(barrier_1);
  }, [barrier_1]);
```
Reason: when user select (tap) some barrier from the list on mobile phone, his choice (`selected_barrier`) shouldn't be wrote in `barrier_1`, because according to design only after clicking on button (called `Select barrier`) we can replace old `barrier_1` value.